### PR TITLE
Filemagic keywords v1

### PR DESCRIPTION
**Pull request by hsadia538**
Describe Changes:

- Add tests for `filemagic` keyword
- Add tests for file_data `prefilter` keyword.

### CODE REVIEW

**Review by jufajardini — CHANGES_REQUESTED** [review]
Congrats on yet another PR :)

I noticed that there are no tests for `file.data` and `prefilter`. As I couldn't find any in our current SV tests, can you please add rules for that, too?

File: tests/file-magic-keywords/test.yaml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+requires:
+  features:
+    - MAGIC
+
+checks:
+    - filter:
+        count: 1
+        match:
+            event_type: alert
+            alert.signature_id: 1
```
[the remark this answers is not in the document]
Comment (hsadia538) [reply tests/file-magic-keywords/test.yaml:10]: We have seperate tests for filestore in other folders. do we still need them here?

**Comment by hsadia538** [comment]
followed by #1127

File: tests/file-magic-keywords/test.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (jufajardini) [reply tests/file-magic-keywords/test.yaml:10]: I usually don't think it's a bad idea to be redundant in this aspect, but I'd say that's not mandatory for this task.

**Pull request closed without merging**